TF_Service_dotNet/TouchFree/TrackingConnectionManager.cs

```diff
@@ -8,20 +9,52 @@ public class TrackingConnectionManager
         public Leap.Controller controller;
         IConfigManager configManager;
 
+        private int maximumWaitTimeSeconds = 30;
+        private int initialWaitTimeSeconds = 1;
+        private int waitTimeSeconds = 1;
+
         public TrackingConnectionManager(IConfigManager _configManager)
         {
             configManager = _configManager;
             controller = new Leap.Controller();
             controller.Connect += Controller_Connect;
+            controller.Disconnect += Controller_Disconnect;
             UpdateTrackingMode(_configManager.PhysicalConfig);
             _configManager.OnPhysicalConfigUpdated += UpdateTrackingMode;
+            CheckConnectionAndRetryOnFailure();
```

Comment:
Is this required when `controller = new Leap.Controller();` exists above? will it not immediately be broken?